File: JAVA/Check for Children Sum Property in a Binary Tree.java
class Solution
{
    public static int isSumProperty(Node root)
    {
        if (root == null) return 1;
        if (root.left != null && root.right != null) {
            if (root.data != (root.left.data+root.right.data)) return 0;
        } else if (root.left != null) {
            if (root.data != (root.left.data)) return 0;     
        } else if (root.right != null) {
            if (root.data != (root.right.data)) return 0;      
        } else {
            return 1;
        }
        int left = isSumProperty(root.left);
        int right = isSumProperty(root.right);
        if (left == 0 || right == 0) {
            return 0;
        } else {
            return 1;
        }
    }
}

File: C++/Check for Children Sum Property in a Binary Tree.cpp
class Solution{
    public:
    int isSumProperty(Node *root)
    {
     if(!root){
         return 1;
     }
     if(!root->left && !root->right){
         return 1;
     }
     int left = isSumProperty(root->left);
     int right = isSumProperty(root->right);
     if(left&&right){
         int sum = 0;
         if(root->left){
             sum+=root->left->data;
         }
         if(root->right){
             sum+=root->right->data;
         }
         return sum==root->data;
     }
     return 0;
    }
};
